web/crux-ui/src/components/registries/edit-registry-card.tsx

```diff
@@ -105,6 +105,19 @@ const EditRegistryCard = (props: EditRegistryCardProps) => {
 
   const registryType = formik.values.type
 
+  const handleRegistryTypeChange = (changedRegistry: RegistryType) => {
+    if (registryType !== changedRegistry) {
+      const registryFields = ['imageNamePrefix', 'url', 'token', 'selfManaged', 'user', '_private']
```

Comment:
@robot9706 Can we replace this with some yup magic? I mean if the type change in the models.ts we will forget this one within a heartbeat.

Reply:
I was also looking at this yesterday. I think yup can't help in this case, but maybe we could extract fields from fromik and use custom attributes to mark them?

Reply:
done, can you pls check again?

Reply:
Looks cool 😎 maybe the `{reset: true}` object could be moved to a const field other than that LGTM

Reply:
Done, thx